lib/babe/verify_test.go

```diff
@@ -697,22 +665,134 @@ func Test_verifier_verifyAuthorshipRight(t *testing.T) {
+	signAndAddSeal(t, kp, headerEquivocatoryPrimary, hashEquivocatoryPrimary[:])
+
+	mockBlockStateEquiv1.EXPECT().GetAllBlocksAtDepth(gomock.Any()).Return([]common.Hash{hashEquivocatoryPrimary})
+	mockBlockStateEquiv1.EXPECT().GetHeader(hashEquivocatoryPrimary).Return(headerEquivocatoryPrimary, nil)
+
+	// Secondary Plain Test Header
+	testParentPrd, err := testBabeSecondaryPlainPreDigest.ToPreRuntimeDigest()
+	assert.NoError(t, err)
+	testParentHeader := newTestHeader(t, *testParentPrd)
+
+	testParentHash := encodeAndHashHeader(t, testParentHeader)
+	testSecondaryPrd, err := testBabeSecondaryPlainPreDigest.ToPreRuntimeDigest()
+	assert.NoError(t, err)
+	testSecPlainHeader := newTestHeader(t, *testSecondaryPrd)
+	testSecPlainHeader.ParentHash = testParentHash
+
+	babeSecPlainPrd2, err := testBabeSecondaryPlainPreDigest.ToPreRuntimeDigest()
+	assert.NoError(t, err)
+	headerEquivocatorySecondaryPlain := newTestHeader(t, *babeSecPlainPrd2)
+
+	hashEquivocatorySecondaryPlain := encodeAndHashHeader(t, headerEquivocatorySecondaryPlain)
+	signAndAddSeal(t, kp, headerEquivocatorySecondaryPlain, hashEquivocatorySecondaryPlain[:])
+	babeVerifier8 := newTestVerifier(t, kp, mockBlockStateEquiv2, scale.MaxUint128, true)
+
+	mockBlockStateEquiv2.EXPECT().GetAllBlocksAtDepth(gomock.Any()).Return([]common.Hash{hashEquivocatorySecondaryPlain})
+	mockBlockStateEquiv2.EXPECT().GetHeader(hashEquivocatorySecondaryPlain).Return(headerEquivocatorySecondaryPlain, nil)
+
+	// Secondary Vrf Test Header
+	encParentVrfDigest := newEncodedBabeDigest(t, testBabeSecondaryVRFPreDigest)
+	testParentVrfHeader := newTestHeader(t, *types.NewBABEPreRuntimeDigest(encParentVrfDigest))
+
+	testVrfParentHash := encodeAndHashHeader(t, testParentVrfHeader)
+	encVrfHeader := newEncodedBabeDigest(t, testBabeSecondaryVRFPreDigest)
+	testSecVrfHeader := newTestHeader(t, *types.NewBABEPreRuntimeDigest(encVrfHeader))
+	testSecVrfHeader.ParentHash = testVrfParentHash
+
+	encVrfDigest := newEncodedBabeDigest(t, testBabeSecondaryVRFPreDigest)
+	assert.NoError(t, err)
+	headerEquivocatorySecondaryVRF := newTestHeader(t, *types.NewBABEPreRuntimeDigest(encVrfDigest))
+
+	hashEquivocatorySecondaryVRF := encodeAndHashHeader(t, headerEquivocatorySecondaryVRF)
+	signAndAddSeal(t, kp, headerEquivocatorySecondaryVRF, hashEquivocatorySecondaryVRF[:])
+	babeVerifierEquivocatorySecondaryVRF := newTestVerifier(t, kp, mockBlockStateEquiv3, scale.MaxUint128, true)
+	mockBlockStateEquiv3.EXPECT().GetAllBlocksAtDepth(gomock.Any()).Return([]common.Hash{hashEquivocatorySecondaryVRF})
+	mockBlockStateEquiv3.EXPECT().GetHeader(hashEquivocatorySecondaryVRF).Return(headerEquivocatorySecondaryVRF, nil)
+
+	tests := []struct {
+		name     string
+		verifier verifier
+		header   *types.Header
+		expErr   error
+	}{
 		{
 			name:     "equivocate - primary",
-			verifier: *babeVerifier7,
-			header:   header7,
+			verifier: *verifierEquivocatoryPrimary,
```

Comment:
This verifier is using a mock that defines `mockBlockStateEquiv1.EXPECT().GetAllBlocksAtDepth(gomock.Any())` but it is possible to replace the `gomock.Any()` with the `headerEquivocatoryPrimary.ParentHash`, same for the others mocks using the `gomock.Any()`

Reply:
:+1:  addressed